app/controllers/pets_controller.rb

```diff
@@ -2,18 +2,16 @@ class PetsController < ApplicationController
 
   def index
     @pets = Pet.all
-    # @shelter = Shelter.find(@pets.shelter_id)
   end
 
   def show
     @pet = Pet.find(params[:id])
-    @shelter = @pet.shelter
   end
 
   def create
-    @shelter = Shelter.find(params[:shelter_id])
-    @pet = @shelter.pets.create(pet_params)
-    redirect_to "/shelters/#{@shelter.id}/pets"
```

Comment:
changing instance variable `@shelter` to `shelter` because local scope is sufficient